File: frontend/src/domain/audio/ports/audio-slicer.ts
/**
 * Audio slicer port.
 */

import { Result } from '@domain/shared/result';
import { AudioSegment } from '../entities/audio-segment.entity';
import { TimeRange } from '../value-objects/time-range';

export interface AudioSlicer {
    slice(
        audioPath: string,
        timeRange: TimeRange,
        outputPath: string
    ): Promise<Result<AudioSegment, Error>>;
}

File: backend/streamcraft/domain/audio/ports/audio_slicer.py
"""Audio slicer port."""

from abc import ABC, abstractmethod
from pathlib import Path

from streamcraft.domain.audio.entities.audio_segment import AudioSegment
from streamcraft.domain.audio.value_objects.time_range import TimeRange
from streamcraft.domain.shared.result import Result


class AudioSlicer(ABC):
    """Port for slicing audio files into segments."""

    @abstractmethod
    def slice(
        self, audio_path: Path, time_range: TimeRange, output_path: Path
    ) -> Result[AudioSegment, Exception]:
        """Slice audio file to time range."""
        ...
